### PR TITLE
refactor(test): Refactor resharding v3 test parameter creation

### PR DESCRIPTION
**Pull request by Trisfald**
A change suggested by @Longarithm a long time ago: using the builder pattern to construct `TestReshardingParameters` in the resharding test.

This should make it easier to change other configurations such as epoch length, accounts, etc.

I added a new crate to autogenerate a large part of the builder struct.

### CODE REVIEW

File: integration-tests/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ bytesize.workspace = true
 borsh.workspace = true
 chrono.workspace = true
 clap.workspace = true
+derive_builder.workspace = true
```
Comment (staffik) [inline integration-tests/Cargo.toml:22]: Does it work if added under `dev-dependencies`? As it is for tests only.

Comment (Trisfald) [reply integration-tests/Cargo.toml]: Yep, correct

**Review by marcelo-gonzalez — APPROVED** [review]

**Comment by Longarithm** [comment]
nit: for the bool parameters, I prefer setters without arguments, like `limit_outgoing_gas()`, `single_shard_tracking()`. The argument is unnecessary information.

**Comment by Trisfald** [comment]
> nit: for the bool parameters, I prefer setters without arguments, like `limit_outgoing_gas()`, `single_shard_tracking()`. The argument is unnecessary information.

I'd love if the crate had an option to do just that, but I couldn't find it 😞

**Pull request merged**